### PR TITLE
PDO

### PR DESCRIPTION
**Pull request by vigneshmanix**
Created pdo class and used it in few lib files

### CODE REVIEW

File: cms/pdo.class.php [hunk 1]
```diff
@@ -0,0 +1,127 @@
+<?php
+if(!defined('__PRAGYAN_CMS'))
+    {
```
Comment (vshriram93) [inline cms/pdo.class.php]: Move to line 2 .

File: cms/pdo.class.php [hunk 2]
```diff
@@ -0,0 +1,127 @@
+<?php
+if(!defined('__PRAGYAN_CMS'))
+    { 
+        header($_SERVER['SERVER_PROTOCOL'].' 403 Forbidden');
+        echo "<h1>403 Forbidden</h1><h4>You are not authorized to access the page.</h4>";
+        echo '<hr/>'.$_SERVER['SERVER_SIGNATURE'];
+        exit(1);
+    }
```
Comment (vshriram93) [inline cms/pdo.class.php]: } , should be the first character. "no whitespace"

File: cms/pdo.class.php [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+/**
+ * @package pragyan
+ * @class pdo Class which uses PHP Data Objects to access database
```
Comment (vshriram93) [inline cms/pdo.class.php]: Add copyright.

File: cms/pdo.class.php [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+ * @author Vignesh Manix
+ * @license http://www.gnu.org/licenses/gpl.html
+ * For more details, see README
+ */
+ 
+class pdodb
+{
```
Comment (vshriram93) [inline cms/pdo.class.php]: '{' should be in the same line as the function or class declaration.

File: cms/pdo.class.php [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    private $pdo;   # @object The PDO object
+    private $stmt;  # @object PDO statement object
+    private $isConnected; # @bool Connected to database
+    private $params; # @array Paramaters of query
+
+
```
Comment (vshriram93) [inline cms/pdo.class.php]: Remove this line.

File: cms/pdo.class.php [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    public function __construct()  # Constructor sets $isConnected, connects to db and creates params array
+                    {
```
Comment (vshriram93) [inline cms/pdo.class.php]: Move this to line 25. like 'public function __construct() {'

File: cms/pdo.class.php [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+                        $this->isConnected = false;
+                        $this->Connect();
+                        $this->params=array();
+                    }
+
+
```
Comment (vshriram93) [inline cms/pdo.class.php]: Remove this line.

File: cms/pdo.class.php [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+                    private function Connect()  # Connect tries to connect 
+                    {
```
Comment (vshriram93) [inline cms/pdo.class.php]: Move it to previous line.

File: cms/pdo.class.php [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+                        if(MYSQL_SERVER=='localhost') $dsn='mysql:dbname='.MYSQL_DATABASE.';host=127.0.0.1';
+                        else $dsn='mysql:dbname='.MYSQL_DATABASE.';host='.MYSQL_SERVER.'';
+                        try
+                            {
```
Comment (vshriram93) [inline cms/pdo.class.php]: Move it to previous line.

File: cms/pdo.class.php [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+                                $this->pdo=new PDO($dsn,MYSQL_USERNAME,MYSQL_PASSWORD,array(PDO::MYSQL_ATTR_INIT_COMMAND => "SET NAMES utf8"));
+                                $this->pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+                                $this->pdo->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
+                                $this->isConnected = true;
+                            }
+                        catch(PDOException $e) 
+                            {
```
Comment (vshriram93) [inline cms/pdo.class.php]: Move it to previous line.

File: cms/pdo.class.php [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+                                echo 'Unhandled Exception. <br />';
+                                echo $e->getMessage();
+                                die();
+                            }
+                    }
+
+
```
Comment (vshriram93) [inline cms/pdo.class.php]: Remove this line.

File: cms/pdo.class.php [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+                    private function Init($query,$params = "") # Init tries to connect, prepare, parameterize, execute and reset params
+                    {
```
Comment (vshriram93) [inline cms/pdo.class.php]: Move it to previous line.

File: cms/pdo.class.php [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+                        if(!$this->isConnected) 
+                            {
```
Comment (vshriram93) [inline cms/pdo.class.php]: Move it to previous line.

File: cms/pdo.class.php [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+                                $this->Connect();
+                            }
+                        try
+                            {
```
Comment (vshriram93) [inline cms/pdo.class.php]: Move it to previous line.

File: cms/pdo.class.php [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+                                $this->stmt=$this->pdo->prepare($query);            
+                                $this->bindMore($params);
+                                if(!empty($this->params))
+                                    {
```
Comment (vshriram93) [inline cms/pdo.class.php]: Fix it in all places.

File: cms/pdo.class.php [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+                    {
+                        $this->isConnected = false;
+                        $this->Connect();
+                        $this->params=array();
+                    }
+
+
+                    private function Connect()  # Connect tries to connect 
+                    {
+                        if(MYSQL_SERVER=='localhost') $dsn='mysql:dbname='.MYSQL_DATABASE.';host=127.0.0.1';
```
Comment (vshriram93) [inline cms/pdo.class.php]: Make sure you have the declaration in the next line. Will be readable

File: cms/authenticate.lib.php [hunk 17]
```diff
@@ -137,9 +139,9 @@ function getGroupIds($userId) {
 	else
 		$groups[] = 1;
 	$groupQuery = 'SELECT `group_id` FROM `' . MYSQL_DATABASE_PREFIX . 'usergroup` WHERE `user_id` = \'' . escape($userId)."'";
-	$groupQueryResult = mysql_query($groupQuery) or die(mysql_error());
-	while ($groupQueryResultRow = mysql_fetch_row($groupQueryResult))
-		$groups[] = $groupQueryResultRow[0];
+	$groupQueryResult = $pdb->query($groupQuery);
```
Comment (nean) [inline cms/authenticate.lib.php]: use binding for queries ?

File: cms/pdo.class.php [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+if (!defined('__PRAGYAN_CMS')) {
+    header($_SERVER['SERVER_PROTOCOL'].' 403 Forbidden');
+    echo "<h1>403 Forbidden</h1><h4>You are not authorized to access the page.</h4>";
+    echo '<hr/>'.$_SERVER['SERVER_SIGNATURE'];
+    exit(1);
+}
+
+class pdodb
+{
+    private $_pdo;   // @object The PDO object
```
Comment (nean) [inline cms/pdo.class.php]: use dockblocks instead of inline comments ?

File: cms/pdo.class.php [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+
+    public function __construct()  // Constructor sets $isconnected, connects to db and creates params array
+    {
+        $this->_isconnected = false;
+        $this->_connect();
+        $this->_params=array();
+    }
+
+
+    private function _connect()  // connect tries to connect
```
Comment (nean) [inline cms/pdo.class.php]: expose connect as public method and pass connection parameters to constructor ?

**Comment by nean** [comment]
expose the pdo instance with a getter

File: cms/authenticate.lib.php (hunk 17 above)
Comment (vigneshmanix) [inline cms/authenticate.lib.php]: @nean In _init function, I'm using prepare. So there is actually no need to escape/quote http://php.net/manual/en/pdo.prepare.php

**Pull request closed without merging**